dropwizard/service/src/main/java/org/apache/polaris/service/dropwizard/auth/PolarisPrincipalRoleSecurityContextProvider.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.polaris.core.auth.PolarisGrantManager;
+import org.apache.polaris.core.context.CallContext;
+import org.apache.polaris.core.context.RealmContext;
+import org.apache.polaris.core.entity.PolarisEntity;
+import org.apache.polaris.core.entity.PrincipalRoleEntity;
+import org.apache.polaris.core.persistence.MetaStoreManagerFactory;
+import org.apache.polaris.core.persistence.PolarisMetaStoreManager;
+import org.apache.polaris.service.auth.BasePolarisAuthenticator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Priority(Priorities.AUTHENTICATION + 1)
+public class PolarisPrincipalRoleSecurityContextProvider implements ContainerRequestFilter {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(PolarisPrincipalRoleSecurityContextProvider.class);
+  @Inject Provider<SecurityContext> securityContextProvider;
+  @Inject MetaStoreManagerFactory metaStoreManager;
+  @Inject Provider<PolarisGrantManager> polarisGrantManagerProvider;
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    AuthenticatedPolarisPrincipal polarisPrincipal =
+        (AuthenticatedPolarisPrincipal) securityContextProvider.get().getUserPrincipal();
+    if (polarisPrincipal == null) {
+      return;
+    }
+    SecurityContext securityContext =
+        createSecurityContext(requestContext.getSecurityContext(), polarisPrincipal);
+    requestContext.setSecurityContext(securityContext);
+  }
+
+  public SecurityContext createSecurityContext(
+      SecurityContext ctx, AuthenticatedPolarisPrincipal principal) {
+    RealmContext realmContext = CallContext.getCurrentContext().getRealmContext();
+    List<PrincipalRoleEntity> activeRoles =
+        loadActivePrincipalRoles(
+            principal.getActivatedPrincipalRoleNames(),
+            principal.getPrincipalEntity(),
+            metaStoreManager.getOrCreateMetaStoreManager(realmContext));
+    Set<String> validRoleNames =
+        activeRoles.stream().map(PrincipalRoleEntity::getName).collect(Collectors.toSet());
+    return new SecurityContext() {
+      @Override
+      public Principal getUserPrincipal() {
+        return principal;
+      }
+
+      @Override
+      public boolean isUserInRole(String role) {
+        return validRoleNames.contains(role);
+      }
+
+      @Override
+      public boolean isSecure() {
+        return ctx.isSecure();
+      }
+
+      @Override
+      public String getAuthenticationScheme() {
+        return ctx.getAuthenticationScheme();
+      }
+    };
+  }
+
+  protected List<PrincipalRoleEntity> loadActivePrincipalRoles(
+      Set<String> tokenRoles, PolarisEntity principal, PolarisMetaStoreManager metaStoreManager) {
+    PolarisCallContext polarisContext = CallContext.getCurrentContext().getPolarisCallContext();
+    PolarisGrantManager.LoadGrantsResult principalGrantResults =
+        polarisGrantManagerProvider.get().loadGrantsToGrantee(polarisContext, principal);
```

Comment:
The PR summary mentions `ActiveRolesProvider`, but I could not find it in the change set... Is it a class/interface?.. Did I miss it?

Reply:
Shoot - a second commit didn't get pushed. I'll fix it